centml/compile.py

```diff
@@ -0,0 +1,98 @@
+import builtins
+from typing import Callable, Dict, List, Optional, Union
+
+import torch
+from prometheus_client import Gauge, start_http_server
+from torch._subclasses.fake_tensor import FakeTensorMode
+
+from centml.compiler.backend import centml_dynamo_backend
+from centml.compiler.config import OperationMode, settings
+from centml.compiler.profiler import Profiler
+
+start_http_server(8000)
+
+
+class Metric:
+    def __init__(self, gpu_name):
+        self._time = 0
+        self._metric = Gauge(f'{gpu_name}_metric', f'Sum of the kernel execution times on {gpu_name}')
+
+    def increment(self, value):
+        self._time += value
+
+    def update_metric(self):
+        self._metric.set(self._time)
+
+    def reset(self):
+        self._time = 0
+        self._metric.set(0)
+
+    def get_value(self):
+        return self._time
+
+
+def compile(
+    model: Optional[Callable] = None,
+    *,
+    fullgraph: builtins.bool = False,
+    dynamic: Optional[builtins.bool] = None,
+    mode: Union[str, None] = None,
+    options: Optional[Dict[str, Union[str, builtins.int, builtins.bool]]] = None,
+    disable: builtins.bool = False,
+) -> Callable:
+
+    # Create a metric for each GPU
+    GPU_METRICS = {gpu: Metric(gpu) for gpu in settings.PREDICTION_GPUS.split(',')}
```

Comment:
We might be able to use labels instead of creating multiple Guages.

https://prometheus.github.io/client_python/instrumenting/labels/

Reply:
We need to check how these metrics are shown in prometheus and grafana. We already have prometheus installed in our local setup. You probably can use that. Or install prometheus and grafana by yourself.